src/internal/agent/hooks/flux.go

```diff
@@ -34,15 +36,34 @@ func NewGitRepositoryMutationHook() operations.Hook {
 func mutateGitRepository(r *v1.AdmissionRequest) (*operations.Result, error) {
 	var patches []operations.PatchOperation
 
+	zarfState, err := getZarfStateFromFileWithinAgentPod(zarfStatePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load zarf state from file: %v", err)
+	}
+
+	// Default to the InCluster gitURL
+	gitServerURL := config.ZarfInClusterGitServiceURL
```

Comment:
This seems strange to me, why wouldn't we just parse the state for the URL like we do for external?  I'd personally rather keep it generic and since this is breaking change anyway, feels like the right time to move to the new pattern.  Is there any benefit to tracking in-cluster git server otherwise?

Reply:
FWIW I had the same thought reviewing this earlier, I just didn't write it down.

Reply:
I'll make this part more generic.

The only other time where it makes sense to track in-cluster git server is when we are creating a read-only user. Since we use Gitea api we can't do that for an external server.